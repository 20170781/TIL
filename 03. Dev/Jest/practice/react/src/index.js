import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './app/app';
import '@fortawesome/fontawesome-free/js/all.js';
import TodoPresenter from './logics/todo_presenter';

const todoPresenter = new TodoPresenter([
  { id: 1, name: 'JavaScript', count: 0 },
  { id: 2, name: 'TypeScript', count: 0 },
  { id: 3, name: 'React', count: 0 },
]);

ReactDOM.render(
  <React.StrictMode>
    <App presenter={todoPresenter} />
  </React.StrictMode>,
  document.getElementById('root')
);
