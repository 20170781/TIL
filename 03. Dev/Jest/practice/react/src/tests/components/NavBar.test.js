import { render } from '@testing-library/react';
import Navbar from '../../components/Navbar';

describe('Navbar', () => {
  test('snapshot', () => {
    const component = render(<Navbar />);

    expect(component.container).toMatchSnapshot();
  });
});
