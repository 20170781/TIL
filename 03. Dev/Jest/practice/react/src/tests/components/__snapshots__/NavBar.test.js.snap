// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<div>
  <div
    class="navbar"
  >
    <i
      class="navbar-logo fas fa-leaf"
    />
    <span>
      Todo App
    </span>
    <span
      class="navbar-count"
      data-testid="total-count"
    />
  </div>
</div>
`;
